### PR TITLE
feat(Piattaforma Notifiche): [IAMVL-84] Add support for PN custom activation flow

### PR DESCRIPTION
**Pull request by emiliopavia**
## Short description
This PR implements the custom activation flow for PN.

https://user-images.githubusercontent.com/467098/180170804-747bf42e-6d97-4178-bfb4-453a4c7323c7.mp4

https://user-images.githubusercontent.com/467098/180172402-377cb0a3-7897-49c3-b3b1-56ca41dfe586.mp4

## List of changes proposed in this pull request
- Show the `PnServiceCTA` when a service has `pn` as `custom_special_flow`
- Add action/reducer/saga/client for the new upsert endpoint
- Add a toggle in the dev settings of the app to switch between test/prod PN environment

## How to test
Use the [IIP-18-add-pn-endpoints](https://github.com/pagopa/io-dev-api-server/tree/IIP-18-add-pn-endpoints) branch of the dev server and add this to your `config.json`:
```
{
  "messages": {
    "pnCount": 1
  },
  "services": {
    "includePn": true
  }
}
```
Set `PN_ENABLED=YES` to `.env` and run the app, then go to services -> Piattaforma delle notifiche digitali -> Avvisi di cortesia and try to enable/disable the service.

### CODE REVIEW

File: ts/components/services/SpecialServices/SpecialServicesCTA.tsx [hunk 1]
```diff
@@ -47,7 +48,8 @@ const SpecialServicesCTA = (props: Props) => {
     boolean
   >([
     ["cgn" as CustomSpecialFlow, isCGNEnabled],
-    ["cdc" as CustomSpecialFlow, isCdcEnabled]
+    ["cdc" as CustomSpecialFlow, isCdcEnabled],
+    ["pn" as CustomSpecialFlow, pnEnabled]
```
Comment (francescopersico) [inline ts/components/services/SpecialServices/SpecialServicesCTA.tsx:52]:
```suggestion
    ["pn" as CustomSpecialFlow, isPnEnabled]
```

Comment (emiliopavia) [reply ts/components/services/SpecialServices/SpecialServicesCTA.tsx]: Done with 9a66aed42dadc083ac3a189fa8b76c7a18b13cae

File: ts/features/pn/api/backendPn.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  method: "post",
+  url: _ => `/api/v1/pn/activation`,
+  headers: composeHeaderProducers(tokenHeaderProducer, ApiHeaderJson),
+  query: query => ({ isTest: query.isTest }),
+  body: body => JSON.stringify(body.pNActivation),
+  response_decoder: upsertPNActivationDecoderCustom()
+};
+
+// client for handling PN endpoints
+export const BackendPnClient = (
```
Comment (francescopersico) [inline ts/features/pn/api/backendPn.ts:50]: Why you are creating another client? You are still talking to the IO Backend.

File: ts/features/pn/store/reducers/activation.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    case getType(pnActivationUpsert.failure):
+      return pot.toError(state, action.payload);
+    case getType(loadServicePreference.request):
+      return pot.none;
+  }
+  return state;
+};
+
+export const pnActivationSelector = createSelector(
+  [(state: GlobalState) => state.features.pn.activation],
```
Comment (francescopersico) [inline ts/features/pn/store/reducers/activation.ts:36]: Is this selector really useful?

File: ts/screens/services/ServiceDetailsScreen.tsx [hunk 4]
```diff
@@ -198,30 +198,31 @@ const ServiceDetailsScreen = (props: Props) => {
...
-                <SpecialServicesCTA
-                  serviceId={props.serviceId}
-                  customSpecialFlow={metadata.custom_special_flow}
-                />
-              </>
-            )}
-          </FooterTopShadow>
-        )}
+        {(maybeCTA.isSome() || SpecialServiceMetadata.is(metadata)) &&
+          canRenderItems && (
```
Comment (dgopsq) [inline ts/screens/services/ServiceDetailsScreen.tsx:202]: This looks a bit convoluted, what do you think about using a variable? Like:
```ts
const showCTA = (maybeCTA.isSome() || SpecialServiceMetadata.is(metadata)) && canRenderItems;
```

File: ts/screens/services/ServiceDetailsScreen.tsx [hunk 5]
```diff
@@ -111,7 +111,7 @@ const ServiceDetailsScreen = (props: Props) => {
 
   useEffect(() => {
     props.loadServiceDetail(props.serviceId);
-  }, []);
+  }, [props]);
```
Comment (francescopersico) [inline ts/screens/services/ServiceDetailsScreen.tsx:114]: Why you want to depend on whole props?

What you think about?

```
  const { loadServiceDetail, serviceId } = props;

  useEffect(() => {
    loadServiceDetail(serviceId);
  }, [loadServiceDetail, serviceId]);
```

**Review by francescopersico — CHANGES_REQUESTED** [review]

**Review by dgopsq — CHANGES_REQUESTED** [review]

File: ts/screens/services/ServiceDetailsScreen.tsx (hunk 5 above)
Comment (emiliopavia) [reply ts/screens/services/ServiceDetailsScreen.tsx:114]: Yep, you'll find this on #4070

File: ts/features/pn/api/backendPn.ts (hunk 2 above)
Comment (emiliopavia) [reply ts/features/pn/api/backendPn.ts:50]: @francescopersico I preferred not to add feature-specific endpoints to the `BackendClient`. It's a pattern I found in-app (e.g. `BackendCGN` or `BackendCdcClient`).

File: ts/screens/services/ServiceDetailsScreen.tsx (hunk 4 above)
Comment (emiliopavia) [reply ts/screens/services/ServiceDetailsScreen.tsx]: Right 12679cb7

**Review by dgopsq — APPROVED** [review]

**Review by francescopersico — APPROVED** [review]

**Pull request merged**